### PR TITLE
Adding optional `controls_transaction?` method

### PR DESCRIPTION
**Pull request by AndrewRayCode**
I tried my hand at writing a test to see if an ActiveRecord migration was created but didn't make much progress. Submitting this for your consideration. Not sure why Travis is saying `NoMethodError: undefined method `sanitize' for ActiveRecord::Base:Class` - I don't get that running locally.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,8 @@ override:
 
 ```ruby
 class NameOfTask < Schlepper::Task
+  USE_TRANSACTION = true
```
Comment (RoryO) [inline README.md]: How about 
```ruby
# Signals to the task runner that this task will control its own transaction.
# When true the task runner will not open a transaction.
# Use with caution.
# @return [Boolean] 
def controls_transaction?
  false
end
```

Then the task runner reads nicely

```ruby
if runner.controls_transaction?
```

File: lib/schlepper/process.rb [hunk 2]
```diff
@@ -75,32 +75,51 @@ def say(*); end
...
-            INSERT INTO schlepper_tasks (version, owner, description, completed_at)
-            VALUES (#{runner.version_number}, #{ActiveRecord::Base.sanitize(runner.owner)}, #{ActiveRecord::Base.sanitize(runner.description)}, #{ActiveRecord::Base.connection.quote(Time.now.to_s(:db))});
-          SQL
-        else
-          puts "#{klass.name} ran without errors, but was not successful"
-          if runner.failure_message
-            puts "The resulting failure was: #{runner.failure_message}"
-          else
-            puts "The failure message was not set. Find #{runner.owner} to help investigate"
+      if use_transaction == false
```
Comment (RoryO) [inline lib/schlepper/process.rb]: Yep this is mostly what I've had in mind for months. Thanks for doing it.

File: lib/schlepper/process.rb [hunk 3]
```diff
@@ -75,32 +75,51 @@ def say(*); end
...
-          fail ActiveRecord::Rollback
         end
       end
 
       puts ''
       puts "Finished #{klass.name}"
       puts '~~~~~~~~~~~~~~~~~~~~~'
     end
+
+    private def run(runner)
```
Comment (RoryO) [inline lib/schlepper/process.rb]: This is a bit confusing with this method named `run` and the task method's `#run`. Perhaps `run_task_for`? Or fix my confusing screwup naming the ivar `runner` to `task`.

**Comment by RoryO** [comment]
Yeah I dunno what happened to travis. It worked for a while and then stopped for some reason.

File: README.md (hunk 1 above)
Comment (RoryO) [reply README.md]: Ah for backwards compatibility you'll need to add `def controls_transaction?` to Schlepper::Task though.

**Comment by AndrewRayCode** [comment]
@RoryO comments addressed and bumped the version. If you want to pair on adding a test that would speed me up, otherwise I'll re-try adding a test later.

**Comment by RoryO** [comment]
You should be able to test for `ActiveRecord::Base.connection.transaction_open?`

Other than that this is great!

**Review by RoryO — APPROVED** [review]

**Pull request merged**